src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/adapter/delius/UserApiService.kt

```diff
@@ -4,25 +4,52 @@ import org.slf4j.LoggerFactory
 import org.springframework.beans.factory.annotation.Qualifier
 import org.springframework.stereotype.Service
 import uk.gov.justice.digital.hmpps.manageusersapi.adapter.WebClientUtils
+import uk.gov.justice.digital.hmpps.manageusersapi.config.DeliusRoleMappings
 import uk.gov.justice.digital.hmpps.manageusersapi.model.DeliusUser
+import uk.gov.justice.digital.hmpps.manageusersapi.model.UserRole
 
 @Service
 class UserApiService(
   @Qualifier("deliusWebClientUtils") val serviceWebClientUtils: WebClientUtils,
+  deliusRoleMappings: DeliusRoleMappings,
 
 ) {
   companion object {
     private val log = LoggerFactory.getLogger(this::class.java)
   }
 
+  private val mappings: Map<String, List<String>> =
+    deliusRoleMappings.mappings.mapKeys { it.key.uppercase().replace('.', '_') }
+
   fun findUserByUsername(username: String): DeliusUser? {
     if ("@" in username) {
       log.debug("Delius not called with username as contained @: {}", username)
       return null
     }
-    return serviceWebClientUtils.getIgnoreError(
-      "/users/$username/details",
-      DeliusUser::class.java,
-    )
+    var users =
+      serviceWebClientUtils.getIgnoreError(
+        "/users/$username/details",
+        DeliusUser::class.java,
+      )
+
+    if (users != null) {
+      users = mapUserDetailsToDeliusUser(users)
+    }
+    return users
   }
```

Comment:
can tidy this up with something similar to:
```
  return serviceWebClientUtils.getIgnoreError
      "/users/$username/details",
      DeliusUser::class.java,
    )?.apply{ mapUserDetailsToDeliusUser(users) }
```

Reply:
Updated